python/app/dialog.py

```diff
@@ -856,3 +848,38 @@ def _on_search_item_selected(self, entity_type, entity_id):
         self.ui.search_input.setText("")
         sg_location = ShotgunLocation(entity_type, entity_id)            
         self._navigate_to(sg_location)
+
+    ##################################################################################################
+    # utility methods
+
+    def _location_from_context(self, ctx):
```

Comment:
does it make sense to move this into `ShotgunLocation`? Now that it's broken out like this, it feels like a factory method to me. `@classmethod ShotgunLocation.from_context()`?